### PR TITLE
Refactor the configuration of the workers

### PR DESCRIPTION
**Pull request by alexduf**
This refactoring:

- isolate the configuration of each worker. Before it was quite the noodle plate
- this should slightly speed up cold starts as we won't be loading the whole config namespace at boot time anymore
- this also give us the possibility to configure the sender workers independently, such that the ios live and the ios edition workers are executing the same code but two different configs.

Now obvisously that means all the config property are being moved, because of the high chance of manual error I wrote a script to automate the migration. This has been successfully ran on CODE. I'll also keep the older properties for a little while, just in case we need to revert.

Here's the migration script (it's not meant to be pretty)
```bash
#!/usr/bin/env bash
[ -z "$STAGE" ] && echo "set STAGE and re-launch" && exit 1;

LOG=$0.log

date | tee -a $LOG

cat conf-migration-mapping.tsv | while read FROM TO ; do
	FROM_WITH_PREFIX="/notifications/${STAGE}/workers/${FROM}"
	TO_WITH_PREFIX="/notifications/${STAGE}/workers/${TO}"
	JSON=$(aws ssm get-parameter --name "${FROM_WITH_PREFIX}" --with-decryption --profile mobile --region eu-west-1 | 
	jq "{Name: \"${TO_WITH_PREFIX}\", Value: .Parameter.Value, Type: .Parameter.Type}")
	echo "${JSON}" >> $LOG
	echo -n Copying $FROM_WITH_PREFIX to $TO_WITH_PREFIX | tee -a $LOG
	VERSION=$(aws ssm put-parameter --overwrite --region eu-west-1 --profile mobile --cli-input-json "$JSON" | jq .Version)
	echo " ✔ (Version: $VERSION)" | tee -a $LOG
done
```

Here's the input file conf-migration-mapping.tsv
```tsv
apns.bundleId	ios/apns.bundleId
apns.certificate	ios/apns.certificate
apns.keyId	ios/apns.keyId
apns.sendingToProdServer	ios/apns.sendingToProdServer
apns.teamId	ios/apns.teamId
mapi.baseUrl	ios/mapi.baseUrl
dryrun	ios/dryrun
cleaner.sqsUrl	ios/cleaningSqsUrl
dryrun	android/dryrun
fcm.debug	android/fcm.debug
fcm.serviceAccountKey	android/fcm.serviceAccountKey
cleaner.sqsUrl	android/cleaningSqsUrl
apns.newsstandBundleId	ios-edition/apns.bundleId
apns.certificate	ios-edition/apns.certificate
apns.keyId	ios-edition/apns.keyId
apns.sendingToProdServer	ios-edition/apns.sendingToProdServer
apns.teamId	ios-edition/apns.teamId
mapi.baseUrl	ios-edition/mapi.baseUrl
dryrun	ios-edition/dryrun
cleaner.sqsUrl	ios-edition/cleaningSqsUrl
dryrun	android-edition/dryrun
fcm.debug	android-edition/fcm.debug
fcm.serviceAccountKey	android-edition/fcm.serviceAccountKey
cleaner.sqsUrl	android-edition/cleaningSqsUrl
cleaner.registration.db.password	cleaner/registration.db.password
cleaner.registration.db.url	cleaner/registration.db.url
cleaner.registration.db.user	cleaner/registration.db.user
cleaner.sqsUrl	cleaner/cleaningSqsUrl
delivery.androidEditionSqsUrl	harvester/androidEditionSqsUrl
delivery.apnsSqsUrl	harvester/iosLiveSqsUrl
delivery.firebaseSqsUrl	harvester/androidLiveSqsUrl
delivery.iosEditionSqsUrl	harvester/iosEditionSqsUrl
registration.db.password	harvester/registration.db.password
registration.db.url	harvester/registration.db.url
registration.db.user	harvester/registration.db.user
registration.db.password	topicCounter/registration.db.password
registration.db.url	topicCounter/registration.db.url
registration.db.user	topicCounter/registration.db.user
topicCounts.bucket	topicCounter/bucket
topicCounts.countThreshold	topicCounter/countThreshold
topicCounts.fileName	topicCounter/fileName
```
Here's a sample output of the script

```
Copying /notifications/CODE/workers/apns.bundleId to /notifications/CODE/workers/ios/apns.bundleId ✔ (Version: 1)
Copying /notifications/CODE/workers/apns.certificate to /notifications/CODE/workers/ios/apns.certificate ✔ (Version: 1)
Copying /notifications/CODE/workers/apns.keyId to /notifications/CODE/workers/ios/apns.keyId ✔ (Version: 1)
Copying /notifications/CODE/workers/apns.sendingToProdServer to /notifications/CODE/workers/ios/apns.sendingToProdServer ✔ (Version: 1)
```

Deployment TODO list
 - [x] apply the migration script to prod
 - [x] manually edit configuration with the new firebase projects for the android-edition
 - [x] merge
 - [ ] check metrics and logs

### CODE REVIEW

File: db-cleaner/cfn.yaml [hunk 1]
```diff
@@ -87,7 +87,7 @@ Resources:
       Handler: Provided
       MemorySize: 128
       Role: !GetAtt ExecutionRole.Arn
-      Runtime: java8
+      Runtime: provided
```
Comment (alexduf) [inline db-cleaner/cfn.yaml]: This is unrelated but now we can finally cloudform a lambda with a provided runtime. That avoids having to manually change the value in the console after applying the value (this is the rust lambda)

File: notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Configuration.scala [hunk 2]
```diff
@@ -46,79 +41,81 @@ case class TopicCountsConfiguration (
...
+    case _ => throw new IllegalStateException("No Platform environment variable defined")
+  }
+
   private def jdbcConfig(config: Config) = JdbcConfig(
     driverClassName = "org.postgresql.Driver",
     url = config.getString("registration.db.url"),
     user = config.getString("registration.db.user"),
     password = config.getString("registration.db.password"),
-    maxConnectionPoolSize = config.getInt("registration.db.maxConnectionPoolSize")
+    maxConnectionPoolSize = 1
```
Comment (alexduf) [inline notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Configuration.scala]: This was of no value to put in the conf. The day we want more connection in a lambda we can edit the code. (but why would we want more than 1?)

File: notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Configuration.scala [hunk 3]
```diff
@@ -46,79 +41,81 @@ case class TopicCountsConfiguration (
...
-    val config = fetchConfiguration()
+    val config = fetchConfiguration(confPrefixFromPlatform)
     ApnsWorkerConfiguration(
-      jdbcConfig(config),
-      config.getString("cleaner.sqsUrl"),
+      config.getString("cleaningSqsUrl"),
       ApnsConfig(
         teamId = config.getString("apns.teamId"),
-        bundleId = if (platform.contains(Ios)) config.getString("apns.bundleId") else config.getString("apns.newsstandBundleId"),
+        bundleId = config.getString("apns.bundleId"),
```
Comment (alexduf) [inline notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Configuration.scala]: Here I use to test what platform we were using (ios or newsstand?) to check which property to load. But now the configuration is prefixed and each platform has its own version of the config.

**Comment by alexduf** [comment]
cc @jimhunty

File: notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Configuration.scala (hunk 2 above)
Comment (NathanielBennett) [reply notificationworkerlambda/src/main/scala/com/gu/notifications/worker/Configuration.scala]: Yeah makes sense

**Comment by alexduf** [comment]
It's live, configuration loads properly and notification are transiting normally

**Pull request merged**